bids/grabbids/bids_validator.py

```diff
@@ -0,0 +1,253 @@
+import os
+import re
+
+__all__ = ['BIDSLayout', 'BIDSValidator']
+
+class BIDSValidator():
+    """An object for BIDS (Brain Imaging Data Structure) verification in a data.
+
+    The main method of this class is `is_bids()`. You should use it for checking whether a file path compatible with BIDS.
+
+    Parameters
+    ----------
+    index_associated : bool, default: True
+        Specifies if an associated data should be checked. If it is true then any file paths in directories `code/`, 
+        `derivatives/`, `sourcedata/`, `stimuli/` and `.git/` will pass the validation, else they won't.
```

Comment:
What's the rationale for making `.git` ever pass? Do we really want to support validation of files in the .git index? I'm not saying that we shouldn't do it this way, just that I'm having trouble seeing a use case where a user would want to including anything in the .git folder in the project index. I can see a lot of cases where a user would want to allow `derivatives`, `stimuli`, etc., but *not* `.git`. So I would incline to remove .git from this list, and force the user to explicitly pass `include=['.git']` if they really want to index files under `.git`.

Reply:
I removed '.git' from associated data (no test yet). I think it is a design decision how desirable associated data should be specified.